content/backend/graphql-python/2-queries.md

```diff
@@ -0,0 +1,172 @@
+
+With an app in place, you need to define a Model – the layer between Django and the database.
+
+<Instruction>
+
+On the `links/models.py` file, add the following content:
+
+```python(path=".../graphql-python/hackernews/links/models.py")
+class Link(models.Model):
+    url = models.URLField()
+    description = models.TextField(null=True, blank=True)
+```
+
+</Instruction>
+
+<Instruction>
+
+Lastly, configure Django to use the new `links` app on the `hackernews/settings.py` file:
+
+```python(path=".../graphql-python/hackernews/hackernews/settings.py")
+INSTALLED_APPS = (
+    # After the graphene_django app
+    'links',
+)
+```
+
+</Instruction>
+
+<Instruction>
+
+Create the database tables:
+
+```bash
+python manage.py makemigrations
+python manage.py migrate
+```
+
+</Instruction>
+
+<Instruction>
+
+Enter the Django shell with the command `python manage.py shell` and create some links:
+
+```bash
+from links.models import Link
+Link.objects.create(url='https://www.howtographql.com/', description='The Fullstack Tutorial for GraphQL')
+Link.objects.create(url='https://twitter.com/jonatasbaldin/', description='The Jonatas Baldin Twitter')
+```
+
+### Creating your first Type and Schema
+In GraphQL, a *Type* is an object that may contain multiple *Fields*. Each field is calculated through *Resolvers*, that returns a value. A collection of types is called a *Schema*. Every schema has a special type called *Query* for getting data from the server and *Mutation* for sending data to the server.
+
+This is a simple overview of the concepts, but it should be enough to go through the tutorial. You can read more about it [here](http://graphql.org/learn/schema/).
+
+<Instruction>
+
+Create the `links/schema.py` file, with the content below:
+
+```python(path=".../graphql-python/hackernews/links/schema.py")
+import graphene
+from graphene_django import DjangoObjectType
+
+from links.models import Link
+
+
+class LinkType(DjangoObjectType):
+    class Meta:
+        model = Link
+
+
+class Query(graphene.AbstractType):
+    links = graphene.List(LinkType)
+
+    @graphene.resolve_only_args
+    def resolve_links(self):
+        return Link.objects.all()
+```
+
+</Instruction>
+
+On the snippet above, the `LinkType` was created using the `DjangoObjectType` – a custom type available in Graphene Django. Also, the special type query was created with a resolver for the field `links`, which returns all the links.
+
+<Instruction>
+
+Create the `hackernews/schema.py` file, with the query type:
+
+```python(path=".../graphql-python/hackernews/hackernews/schema.py")
+import graphene
+
+import links.schema
+
+
+class Query(links.schema.Query, graphene.ObjectType):
+    pass
+
+
+schema = graphene.Schema(query=Query)
+```
+
+</Instruction>
+
+Why creating another query? This query just inherits the query defined before. This way, you are able to keep every part of the schema isolated in the apps. After, the schema is defined, with the main query.
```

Comment:
Yeah, I don't know why I put this last sentence – "After, the schema..." – removed it to make more sense.